Program.cs

```diff
@@ -12,31 +12,42 @@ static void Main(string[] args)
 			if (args.Length<2) return;
 			var host = args[0];
 			var port = (short)int.Parse(args[1]);
-			Console.Write($"Connecting to {host}:{port}");
-			Console.WriteLine( TryConnect(host, port)?" OK":" NOK");
+			Console.WriteLine($"Connecting to {host}:{port}");
+            var response = TryConnect(host, port);
+            var startColor = Console.ForegroundColor;
+            if (string.IsNullOrEmpty(response))
+            {
+                Console.ForegroundColor = ConsoleColor.Green;
+                Console.WriteLine("OK");
+            }
+            else
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine(response);
+            }
+            Console.ForegroundColor = startColor;
         }
 		
-		static bool TryConnect(string host, short port)
+		static string TryConnect(string host, short port)
 		{
-		Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+		    Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
 
-// Connect using a timeout (5 seconds)
+            // Connect using a timeout (5 seconds)
 
-IAsyncResult result = socket.BeginConnect( host, port, null, null );
+            IAsyncResult result = socket.BeginConnect( host, port, null, null );
 
-bool success = result.AsyncWaitHandle.WaitOne(1000, true);
+            result.AsyncWaitHandle.WaitOne(1000, true);
 
-if (socket.Connected)
-{
-	socket.EndConnect(result);
-	return true;
-}
-else
-{
-	// NOTE, MUST CLOSE THE SOCKET
-
-	socket.Close();
-	return false;
-}}			
+            if (socket.Connected)
+            {
+	            socket.EndConnect(result);
+	            return "";
+            }
+            else
+            {
+                socket.Close();
+                return "FAIL";
```

Comment:
Going from "bool" (consistent with the c# standard response for `Try...` methods) to string with "" resp "FAIL" seems like a regression to me.

IF this method should be extended (for another feature) with more failure info, I think it should return a Union type of <OK or ErrorMessage> to be clearer. In any case, not part of this PR